### ISSUE TITLE
Problems with pbo

### ISSUE THREAD

**Issue by Whitemanns** [RaJiska/JAPM#2]
Hello I know this has not been updated in a while but I am currently using a Raspberry Pi to decompile so I can edit it recompile back into the PBO but when I put it into the server it states that the sizes too small. Thank you for your time

**Comment by RaJiska** [comment]
Hello,

Could you provide the PBO file which you have your issue with ?

**Comment by Whitemanns** [comment]
It is a PBO for Arma 3 Altis life

**Comment by RaJiska** [comment]
I'd need you to attach the file to a comment or provide a direct link, this would ensure I have the exact same file you used to when you experienced your issue.

**Comment by Whitemanns** [comment]
https://drive.google.com/file/d/1iE44j_a0HPWUtFNQIgU6Lxp706u93H92/view?usp=drivesdk

**Comment by RaJiska** [comment]
I've been looking through it and appears your PBO does not follow the [format standard](https://community.bistudio.com/wiki/PBO_File_Format#PBO_Header_Entry). After looking closer to your archive, it appears for each entry there is a shift of 8 bits in entries.

You can solve this by replacing the structure `pbo_entry_meta_t` in the file `pbo.h` with the following one:

```
typedef struct __attribute__((__packed__))
{
	uint32_t packing;
	uint32_t original_size;
	uint16_t garbage;
	uint8_t garbage2;
	uint32_t timestamp;
	uint32_t data_size;
} pbo_entry_meta_t;
```

Beware though as this may break compatibility with other PBO files.

**Comment by Whitemanns** [comment]
I'm getting the same are set with and I'll post that down below. So when I use a Windows computer the file comes out to just under 18 megabytes but when I use the pi it shows that it's compressing it down to 16 megabytes and then this error shows stating that this file size should be 1GB which makes no sense because the files never been that big.

18:31:06 ErrorMessage: Data file too short 'mpmissions\Altis_Life.Altis.pbo'. Expected -1274963735 B, got 17674245 B18:31:06 Application terminated intentionallyErrorMessage: Data file too short 'mpmissions\Altis_Life.Altis.pbo'. Expected -1274963735 B, got 17674245 B

**Comment by RaJiska** [comment]
Ah this is indeed another issue; I don't understand how you managed to get it unarchived first (assuming you used the exact same PBO you sent me; are you sure you send me the original PBO ?) as this was failing on my end; the fix I provided was meant to allow the unarchiving.

Unfortunately I won't be able to be of much help regarding the rearchiving as I don't have my Arma 3 development setup anymore. Funny how this expects a negative value, it looks like an overflow to me and may be linked to the 8 bits shifting bug I noticed earlier. Without applying the fix I provided in my previous post, the PBO reports the `mission.sqm` file to be several hundreds of GBs; this may overflow the Arma 3 server.
